### PR TITLE
Capitalize first letter of folder name similar to web client

### PR DESCRIPTION
**Pull request by sudhirkhanger**
ownCloud web client when creating a folder would capitalize first letter of the folder name whereas Android app creates a folder name with small letters. `textCapSentences` in the `inputType` will do the same on Android.

I messed up git commands and pushed my changed to pull mentioned below.

https://github.com/owncloud/android/pull/1488
https://github.com/owncloud/android/issues/1495

### CODE REVIEW

**Comment by przybylski** [comment]
This one looks ok to me.
please squash commits to a single one and force push to avoid merging conflicts

**Comment by sudhirkhanger** [comment]
Hi @przybylski,

I have squashed and force pushed the commits. Thanks for looking.

Sudhir.

File: res/layout/edit_box_dialog.xml [hunk 1]
```diff
@@ -18,19 +18,17 @@
...
 -->
 <LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
-    android:layout_width="match_parent"
-    android:layout_height="match_parent"
-    android:gravity="clip_horizontal"
-    android:orientation="vertical" >
+              android:layout_width="match_parent"
+              android:layout_height="match_parent"
+              android:gravity="clip_horizontal"
+              android:orientation="vertical">
```
Comment (tobiasKaminsky) [inline res/layout/edit_box_dialog.xml]: There is no change but only code formatting.
Would be good if this is not in the commit.

Comment (sudhirkhanger) [inline res/layout/edit_box_dialog.xml]: The only change is adding `textCapSentences` in `android:inputType="textNoSuggestions|textCapSentences"`.

> Would be good if this is not in the commit.

@tobiasKaminsky Why do you mean?

Comment (AndyScherzinger) [inline res/layout/edit_box_dialog.xml]: according to the diff above the only change within the xml file is the padding of four lines. So there is no change except for file formatting so @tobiasKaminsky proposes to remove this change.

Comment (sudhirkhanger) [inline res/layout/edit_box_dialog.xml]: @AndyScherzinger @tobiasKaminsky Do you want to me to remove formatting and only make required changes?

If that is the case then do I revert the commit, make changes and push again. Right?

Sorry for taking up so much time. I hope I can improve in this area.

Comment (AndyScherzinger) [inline res/layout/edit_box_dialog.xml]: That would be great! :)
No worries @sudhirkhanger we've all been there with our first contributions :+1:

Comment (przybylski) [inline res/layout/edit_box_dialog.xml]: Personally i don't see this as a problem, you should always leave code in better shape than you stumble upon it. Even if it is simple reformatting for readability.

Comment (AndyScherzinger) [inline res/layout/edit_box_dialog.xml]: True, but as long as not all people use the same formatter, resolving conflicts is a real pain... :)

**Comment by sudhirkhanger** [comment]
@tobiasKaminsky @przybylski @AndyScherzinger 

Please take a look. It should be as required.

File: res/layout/edit_box_dialog.xml [hunk 2]
```diff
@@ -28,7 +28,7 @@
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
         android:ems="10" 
-		android:inputType="textNoSuggestions"
+		android:inputType="textNoSuggestions|textCapSentences"
 		>
```
Comment (AndyScherzinger) [inline res/layout/edit_box_dialog.xml]: Looks good to me, I would shorten the XML using /> instead of a complete closing tag though :+1:

Comment (sudhirkhanger) [inline res/layout/edit_box_dialog.xml]: Done.

**Comment by AndyScherzinger** [comment]
@davivel @tobiasKaminsky looks good to me :+1:

**Comment by przybylski** [comment]
Now this file has mixed spaces and tabs :P

**Comment by tobiasKaminsky** [comment]
Yeah, the intent of the last line is somehow wrong.
Sorry to bother you with this during your first commit :/

**Comment by tobiasKaminsky** [comment]
Oh, btw: as you are a member of ownloud dev you can push the branch directly to the owncloud app repository.
This will make it much more easier for others to do small changes.

**Comment by sudhirkhanger** [comment]
@tobiasKaminsky How does that work? I create a branch locally, push it to origin(my fork), finish my work in origin, and then push the branch to upstream.

    git fetch upstream
    git checkout -b my_new_master_branch upstream/master
    git push -u origin my_new_master_branch

Then I do my work. And once finished then I can `git push upstream my_new_master_branch`.

PS: I have corrected the indent problem.

**Comment by tobiasKaminsky** [comment]
I always push my changes directly to an upstream branch.
And then create a PR from upstream:myNewBranch.

You have not done your last mentioned step: git push upstream my_new_master_branch?

**Comment by sudhirkhanger** [comment]
@tobiasKaminsky Looks like I don't have permissions to do that. I have though replaced the indents with spaces and hopely PR can be accepted.

```
$ git push upstream capitalize_folder_name
ERROR: Permission to owncloud/android.git denied to sudhirkhanger.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

**Comment by tobiasKaminsky** [comment]
Strange. @davivel can you check his permissions?

**Comment by przybylski** [comment]
@sudhirkhanger whats your
`git remote -v`
?

**Comment by sudhirkhanger** [comment]
@przybylski 

```
$ git remote -v
origin  git@github.com:sudhirkhanger/android.git (fetch)
origin  git@github.com:sudhirkhanger/android.git (push)
upstream        git@github.com:owncloud/android.git (fetch)
upstream        git@github.com:owncloud/android.git (push)
```

**Comment by davivel** [comment]
Hi, @sudhirkhanger .

> ownCloud web client when creating a folder would capitalize first letter of the folder name

Sorry, I am confused. The web client doesn't do that, the letters are kept as the user enters them.

Is it possible that your web browser does that due to some configuration option?

**Comment by davivel** [comment]
@karlitschek , could you check the write permissions of @sudhirkhanger for this repo? Seems he already sent the contribution agreement in the past.

**Comment by karlitschek** [comment]
@sudhirkhanger Sorry. You should have access now. @davitol FYI

**Comment by davitol** [comment]
@davivel

**Comment by sudhirkhanger** [comment]
@davivel,

@jancborchardt in post no. 4 in the thread linked below indicates that it is indeed the default behavior of ownCloud web-client.

https://github.com/owncloud/android/issues/1495

**Comment by jancborchardt** [comment]
Yeah, +1 on capitalizing the first letter.

**Comment by davivel** [comment]
@sudhirkhanger , thanks a lot for the reference to #1495, it was really helpful to understand the context :) 

@jancborchardt , I'm not sure we are all talking about the same element. @sudhirkhanger is capitalizing the **user input** , this is, enforcing the folder name entered by the user to start with a capital letter. Are you supporting this, or supporting that the **dialog title** is capitalized in the first letter?

**Comment by jancborchardt** [comment]
Just for the record cc @davivel: I’m fine with this if it is basically just by default enabling the shift key for the first letter in the input field, and people can tap it to start with a small letter.

I’m _not_ ok with this if it takes whatever was put in and forces the first letter to be uppercase.

**Comment by davivel** [comment]
Yes, maybe it's my bad, I was assuming that it doesn't allow the user to change first later to lowercase, but the plain truth is that I didn't test it... I should have, sorry :$

So I tested now, and the problem is different. It's just not working, no capitalization at all. I used a Nexus 5 with Android 6 for the test. 

@sudhirkhanger , what device are you using to test? What is your Android version?

**Comment by sudhirkhanger** [comment]
@davivel It only turns on shift for the first letter of a sentence. You can turn it off if you want your folders to have lowercase.

It seems to work fine on Moto G 3rd Gen running Android 6.0.1 and x86 emulator running minSDK level. Please see the screencast below.

https://youtu.be/4G9Gu02oVnA

I am traveling for next few days. Once I am back I will try with more emulator images just to make sure it's working.

**Comment by davivel** [comment]
Awesome demo! Thanks a lot.

Please, don't put much time in testing with different emulators. There are too many differences in the behaviour of emulators and real devices when we talk about the ```android:inputType```. In this case, emulator are not a good reference.

**Comment by malkomich** [comment]
I have been testing the fix and it seems to be working. 

When the keyboard settings allow the auto capitalization the input text can receive the capitalize text, but if this setting is disabled it is not forced to be capitalized, so this is the expected behaviour.

So...it seems OK for me :+1:

**Comment by davivel** [comment]
Thanks, @malkomich . That explains why it didn't work in my fast test, my keyboard app has the capitalization disabled.

Ready to be tested, then.

**Comment by davivel** [comment]
After internal discussion, taking into account the functional tests already done by @malkomich and the scope of this PR, there's no need to create a specific test plan for this, QA is considered as OK.

cc @rperezb @jesmrec 

I'll rebase & merge.

**Comment by davivel** [comment]
Better: @sudhirkhanger , could you rebase your branch capitalize_folder_name on upstream/master , the push with -f option? Then I will merge.

If you have problems rebasing tell me here.

Thanks.

**Comment by sudhirkhanger** [comment]
@davivel I am unable to squash this into a single commit. The repos should be updated now.

Thanks.

**Comment by davivel** [comment]
Thanks, @sudhirkhanger . I'll take care when the tests finish.

**Comment by malkomich** [comment]
@sudhirkhanger  Have you the remote repo updated (git fetch) ?

**Comment by sudhirkhanger** [comment]
@malkomich Yes the remote repo should be updated. `This branch is 5 commits ahead, 11 commits behind owncloud:master.`

**Comment by malkomich** [comment]
I have replicate the PR here: https://github.com/owncloud/android/pull/1645, to get it rebased on master successfully and be able to merge ASAP.

For next time @sudhirkhanger, we use "rebase" instead of "merge". If you want to check a comparison of both methods you can do it here:
https://www.atlassian.com/git/tutorials/merging-vs-rebasing/conceptual-overview
https://www.atlassian.com/git/articles/git-team-workflows-merge-or-rebase/

Let's keep your contribution here: https://github.com/owncloud/android/pull/1645 . Thanks :smile:

**Pull request closed without merging**